packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -1043,6 +1049,28 @@ enum NullSafetyMode {
   autodetect,
 }
 
+/// Indicates the module system DDC is targeting.
+enum DdcModuleFormat {
+  amd,
+  ddc,
+  es6,
+}
+
+DdcModuleFormat? _ddcModuleFormatFromFrontEndArgs(List<String>? extraFrontEndArgs) {
```

Comment:
Rather than parsing extra front end args, can we instead add a top-level flutter tool option `--dartdevc-module-format`? I'm assuming we would only need this from `flutter run`, in which case you can just add the option in the constructor to `RunCommandBase`, around here: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/run.dart#L46

I'm guessing that the CFE needs to know about this also. If this is the case, I would recommend checking for the flutter tool option you added and then forwarding it to extra frontend options around here: https://github.com/Markzipan/flutter/blob/legacy-support/packages/flutter_tools/lib/src/runner/flutter_command.dart#L1212

Reply:
This is fine by me. However, would this require us to have a deprecation process for this new flag? Our intent is that this migration isn't user-visible. The flag would only be used by Dart Web developers or tooling.

Reply:
To summarize an offline discussion:

1. we do not expect anyone other Flutter/Dart maintainers to even know about the `--dartdevc-module-format=` frontend flag
2. we want the module format to be temporarily configurable while the new feature is iterated on. once it is landed, amd support will be removed.

I'm fine with leaving this function here then, if you add a TODO to remove it once deprecate AMD support.

Reply:
Sounds good - updated!